src/features/presence/utils/presenceUtils.js

```diff
@@ -60,4 +64,24 @@ function determineInteractionMessage(interactionType, interactedAt) {
     return message;
 }
 
-export { sortByActivity, getLastActionTimeMS, determineInteractionMessage };
+function convertMillisecondsToUnitAndValue(ms) {
+    const seconds = ms / 1000;
+    const abs = Math.abs(seconds);
+    if (abs < SEC_IN_MINUTE) {
+        return { unit: 'second', value: seconds };
+    }
+    if (abs < SEC_IN_HOUR) {
+        return { unit: 'minute', value: Math.round(seconds / SEC_IN_MINUTE) };
+    }
+    if (abs < SEC_IN_DAY) {
+        return { unit: 'hour', value: Math.round(seconds / SEC_IN_HOUR) };
+    }
+
+    if (abs < SEC_IN_MONTH) {
```

Comment:
months don't share the same amount of days so this calculation wouldn't be technically accurate. is that okay? is this behavior different than the react-intl 2.x behavior?